packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -83,6 +83,7 @@ class ChangeNotifier extends Listenable {
   /// [notifyListeners]; and similarly, by overriding [removeListener], checking
   /// if [hasListeners] is false after calling `super.removeListener()`, and if
   /// so, stopping that same work.
+  @visibleForTesting
```

Comment:
why?

Reply:
I test that the listeners are indeed removed when I remove the builder from the tree

Reply:
i would just subclass ChangeNotifier in your test. But this is fine too I guess.